src/services/persistent-storage.service.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { PersistentStorage } from "./persistent-storage.service";
+import { mock, mockReset } from "jest-mock-extended";
+import { defer } from "../utils/HelperFunctions";
+
+describe("PersistentStorage Unit Tests", () => {
+  let storage: Storage;
+  let persistentStorage: PersistentStorage;
+
+  beforeAll(() => {
+    storage = mock<Storage>();
+    persistentStorage = new PersistentStorage("testing", storage);
+  });
+
+  afterEach(async () => {
+    // Reset the mock after the JS stack has cleared
+    await defer(() => {});
+    mockReset(storage);
+    await persistentStorage.clear();
+    await defer(() => {});
+  });
```

Comment:
We should consider `beforeEach` instead of `beforeAll`. This will remove the need for clearing the storage each time

Reply:
I used it like this because I found that `mock<Storage>()` is actually a singleton when I wrote the `init` test.

Reply:
Just to make sure I understand what you mean here are we saying the following is true:
```ts
const mock1 = mock<Storage>();
const mock2 = mock<Storage>();

mock1 === mock2 // true
```

Reply:
Yes

Reply:
That's quite alarming to say the least. Surely there's a way around that? I'd be reluctant to use it just based on that. Easily become a source of false positives in tests